pywarp10/gts.py

```diff
@@ -46,167 +46,55 @@ def is_gts(x: Any) -> bool:
-                labels += f"  {key}={value}\n"
-        else:
-            labels = ""
-        if self.attributes:
-            attributes = "attributes:\n"
-            for key, value in self.attributes.items():
-                attributes += f"  {key}={value}\n"
-        else:
-            attributes = ""
-        if "timestamps" in self.data.columns:
-            data = self.data[["timestamps", "values"]].__repr__()
-            return f"{name}{labels}{attributes}\n\n{data}"
-        else:
-            return f"{name}{labels}{attributes}\n\nEmpty GTS"
-
-    def __eq__(self, other: "GTS") -> bool:
-        """Check if two GTS are equal.
-
-        Args:
-            other: The other GTS to compare with.
-
-        Returns:
-            True if the two GTS are equal.
-        """
-        if isinstance(other, GTS):
-            if (
-                self.classname == other.classname
-                and self.labels == other.labels
-                and self.attributes == other.attributes
-                and self.data.equals(other.data)
-            ):
-                return True
-        return False
+class GTS(pd.DataFrame):
+    def __init__(self, data: Union[Dict, List]):
+        if isinstance(data, dict):
+            data = self._init_gts(data)
+        elif isinstance(data, list):
+            data = self._init_lgts(data)
+        # Convert to timestamps only if higest timestamp is greater than 1 day after the
+        # epoch
+        if len(data.index) > 0 and max(data.index) > 86400000000:
+            data.index = pd.to_datetime(data.index, unit="us")
+        super().__init__(data)
 
-
-class LGTS(pd.DataFrame):
-    def __init__(self, l: List) -> None:
-        lgts = []
-        for x in l:
-            if not is_gts(x):
-                raise TypeError("The list is not a list of GTS.")
-            lgts.append(GTS(x).to_pandas())
-        res = pd.concat(lgts)
-        res.replace("", float("NaN"), inplace=True)
-        res.dropna(how="all", axis=1, inplace=True)
-        super().__init__(res)
-
-    def __eq__(self, other: "LGTS") -> bool:
-        """Check if two LGTS are equal.
+    def _init_gts(self, gts: Dict) -> None:
+        """Initialize the GTS.
 
         Args:
-            other: The other LGTS to compare with.
-
-        Returns:
-            True if the two LGTS are equal.
+            data: The data to initialize the GTS with.
         """
-        if isinstance(other, LGTS) and self.equals(other):
-            return True
-        return False
-
-    @staticmethod
-    def from_dataframe(
-        x: pd.DataFrame, timestamp_col="timestamps", value_col="values"
-    ) -> List:
-        """Transform a dataframe to warpscript.
+        if not is_gts(gts):
+            raise TypeError(f"{gts} is not a GTS")
+        if "c" in gts.keys():
+            gts["classname"] = gts.pop("c")
+            gts["timestamps"] = [d[0] for d in gts["v"]]
+            gts["values"] = [d[1] for d in gts["v"]]
```

Comment:
wroks with multivalues?

Reply:
Not for now, I didn't plan to implement it yet.